contributors/design-proposals/service-injection-policy.md

```diff
@@ -0,0 +1,404 @@
+# Service Injection Policy
+
+  * [Abstract](#abstract)
+  * [Motivation](#motivation)
+  * [Constraints and Assumptions](#constraints-and-assumptions)
+  * [Use Cases](#use-cases)
+    * [Loose coupling between services and their consumers](#loose-coupling-between-services-and-their-consumers)
+  * [Proposed Changes](#proposed-changes)
+    * [ServiceInjectionPolicy API object](#serviceinjectionpolicy-api-object)
+    * [AdmissionControl Plugin: ServiceInjectionPolicy](#admissioncontrol-plugin-serviceinjectionpolicy)
+      * [Behavior](#behavior)
+      * [Optional Behavior with implicit injections](#optional-behavior-with-implicit-injections)
+  * [Examples](#examples)
+    * [Simple Pod Spec Example](#simple-pod-spec-example)
+    * [Pod Spec with `ConfigMap` Example](#pod-spec-with-`configmap`-example)
+    * [ReplicaSet with Pod Spec Example](#replicaset-with-pod-spec-example)
+
+
+## Abstract
+
+Describes a policy resource that allows loose coupling from services to the pods
+that consume them.
+
+## Motivation
+
+Consuming a service involves more than just connectivity.  In addition to
+coordinates to reach the service, credentials and non-secret configuration
+parameters are typically needed to use the service.  The primitives for this
+already exist, but a gap exists where loose coupling is desired: it should be
+possible to inject pods with the information they need to use a service on a
+service-by-service basis, without the pod authors having to incorporate the
+information into every pod spec where it is needed.
+
+## Constraints and Assumptions
+
+1.  New mechanisms must be made to work with controllers such as deployments and
```

Comment:
> Existing controllers that create pods should recreate their pods when a new Service Injection Policy is added that would effect them.

I am not sure this is always true.  Imagine a new SIP is created that targets all Pods, this would then restart all Pods in the namespace.  The impact could be limited by adding a PodDisruptionBudget, but that part of the design would need to be further discussed.  May be we should defer this and leave it as a possible future work.